src/read_write/ply.rs

```diff
@@ -408,48 +407,117 @@ impl PlyIterator {
             reader: BufReader::with_capacity(num_bytes_per_point * 1024, file),
             readers,
             num_total_points: header["vertex"].count,
+            batch_size,
             offset: header.offset,
             point_count: 0,
         })
     }
 }
 
+fn to_vec<T>(slice: &[u8], cur_batch_size: usize) -> Vec<T>
+where
+    T: Clone,
+{
+    let ptr = slice.as_ptr() as *const T;
+    unsafe { std::slice::from_raw_parts(ptr, cur_batch_size).to_vec() }
```

Comment:
What do you think about omitting a copy here by using https://doc.rust-lang.org/std/vec/struct.Vec.html#method.from_raw_parts and std::mem::replace?

Reply:
I changed the implementation to use `AttributeData`, so we don't need any `unsafe` methods any more.